pilot/pkg/xds/rds.go

```diff
@@ -58,6 +58,8 @@ func rdsNeedsPush(req *model.PushRequest) bool {
 }
 
 func (c RdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *model.WatchedResource, req *model.PushRequest) model.Resources {
+	proxy.Lock()
```

Comment:
why do we need this lock

Reply:
race without it

Reply:
Then i doubt the meaning of move it to the receive go routine